snipssonos/services/spotify/music_custom_service.py

```diff
@@ -0,0 +1,38 @@
+import json
+
+from snipssonos.entities.artist import Artist
+from snipssonos.helpers.spotify_client import SpotifyClient, SpotifyAPISearchQueryBuilder
+
+
+class SpotifyCustomService:
+    TIME_RANGES = ["long_term", "medium_term", "short_term"]
+
+    def __init__(self, client_id, client_secret, access_token=None, refresh_token=None):
+        self.client = SpotifyClient(client_id, client_secret, access_token, refresh_token)
+        self.client_id = client_id
+        self.client_secret = client_secret
+
+        self.client.set_user_endpoint()
+
+    def fetch_top_artist(self):
+        self.client.set_top_artist_endpoint()
+        artists_results = []
+        for time_range in self.TIME_RANGES:
+            top_artist_by_time_range_query = SpotifyAPISearchQueryBuilder(is_request_user_data_query=True)
```

Comment:
I think it would be better to have a builder that goes like this : 

```
top_artist_by_time_range_query = SpotifyAPISearchQueryBuilder()
     .request_user_data_query()
     .add_time_range(time_range)
```

Reply:
Agreed. Works for me!